client/lib/importer/importer-config.js

```diff
@@ -186,7 +186,7 @@ function getConfig( { siteTitle = '' } = {} ) {
 				args: { exampleUrl: 'https://newsletter.substack.com/' },
 			} ),
 			validate: ( urlInput ) => {
-				return /^https:\/\/[\w-]+\.substack\.com\/?$/.test( urlInput.trim() );
+				return /^(https):\/\/[^ "]+$/.test( urlInput.trim() ) && urlInput.length <= 255;
```

Comment:
While the `.substack.com` check clearly needs to go, can the `validate` function just be removed? The only other importer that takes a URL is Wix, and that has no `validate` function.

This validation would show a non-Substack URL as being correct. We could potentially scrape the URL to see if it's a Substack site, but I don't think it'd be a huge benefit, it's more straightforward to remove the check, instead.

Reply:
I've done this in the latest commit. This does mean folks can get away with entering anything here.

<img width="702" alt="CleanShot 2021-08-05 at 15 43 54@2x" src="https://user-images.githubusercontent.com/533/128333732-6a244337-5b56-41b7-80e4-cdbe8c281c39.png">

Reply:
I agree it's not ideal. It might be worth gathering data on whether folks actually submit invalid URLs, to see if we need to implement more comprehensive checks.